src/Optimization/hiopResidual.cpp

```diff
@@ -58,11 +58,13 @@ hiopResidual::hiopResidual(hiopNlpFormulation* nlp_)
 {
   nlp = nlp_;
   rx = nlp->alloc_primal_vec();
+  rx->setToZero();
   rd = nlp->alloc_dual_ineq_vec();
-  rxl = rx->alloc_clone();
-  rxu = rx->alloc_clone();
-  rdl = rd->alloc_clone();
-  rdu = rd->alloc_clone();
+  rd->setToZero();
+  rxl = rx->new_copy();
+  rxu = rx->new_copy();
+  rdl = rd->new_copy();
+  rdu = rd->new_copy();
```

Comment:
I also fixed this in my fr branch. I think it is better to PR it here and then I will rebase my fr to the develop branch.

Reply:
ok, will merge